src/git/zcl_abapgit_git_transport.clas.abap

```diff
@@ -17,6 +17,26 @@ CLASS zcl_abapgit_git_transport DEFINITION
         !eo_branch_list  TYPE REF TO zcl_abapgit_git_branch_list
       RAISING
         zcx_abapgit_exception .
+    CLASS-METHODS upload_pack_by_commit
+      IMPORTING
+        !iv_url          TYPE string
+        !iv_hash         TYPE zif_abapgit_definitions=>ty_sha1 OPTIONAL
+        !iv_deepen_level TYPE i DEFAULT 0
+      EXPORTING
+        !et_objects      TYPE zif_abapgit_definitions=>ty_objects_tt
+        !ev_commit       TYPE zif_abapgit_definitions=>ty_sha1
+      RAISING
+        zcx_abapgit_exception .
+    CLASS-METHODS upload_pack
```

Comment:
`upload_pack_by_branch` and `upload_pack_by_commit` and `upload_pack` as public, when should which be used for what?

Reply:
sry, missed that - of coure you're right - will make upload_pack private